File: 125-valid-palindrome/valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        StringBuilder res=new StringBuilder();
        for(char ch:s.toCharArray()){
            if(Character.isLetterOrDigit(ch)){
                res.append(Character.toLowerCase(ch));
            }
        }
        int l=0,r=res.length()-1;
        while(l<r){
            if(res.charAt(l)!=res.charAt(r)){
                return false;
            }
            l++;
            r--;
        }
        return true;
    }
}

File: 125-valid-palindrome/valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        s=s.lower()
        t=""
        for ch in s:
            if ch.isalnum():
                t+=ch
        i=0
        j=len(t)-1
        while(i<j):
            if t[i]!=t[j]:
                return False
            i+=1
            j-=1
        return True